### PR TITLE
Feature/persist the store

### LINKED ISSUE #97: Persist store
https://docs.pmnd.rs/zustand/integrations/persisting-store-data

### PR DESCRIPTION
**Pull request by tomivm**
Persist the Zustand store using the Zustand persist middleware
Adapt the board UI to show a loader during hydration
close #97

### CODE REVIEW

File: src/stores/board-slice.ts
Comment (RodriSanchez1) [inline src/stores/board-slice.ts:1]: I noticed the variable names `showInitialContent` and `displayInitialContent`, and I believe renaming them could enhance clarity and maintainability in our codebase.

`displayInitialContent`: This variable's name suggests an action to display content, but it actually represents a boolean state. To accurately reflect its purpose, consider renaming it to something like `isInitialContentVisible` or `initialContentVisible`. By renaming it, we make it clear that this boolean variable represents the visibility state of the initial content.

`showInitialContent`: Similarly, this variable name could be more descriptive. Renaming it to `setInitialContentVisible` or `toggleInitialContentVisibility` would clarify its role in setting the visibility state of the initial content.

Choosing more descriptive variable names can help reduce ambiguity and make our codebase easier to understand for both current and future developers. - CHAT GPT

File: src/providers/StoreProvider.tsx
Comment (RodriSanchez1) [inline src/providers/StoreProvider.tsx:1]: Well donee!

**Review by RodriSanchez1 — CHANGES_REQUESTED** [review]
The prompt Form its blinking after change the prompt and press the NEW BOARD button. I think that is not an excepected behaviour

File: src/stores/board-slice.ts
Comment (tomivm) [reply src/stores/board-slice.ts:1]: Good approach! I renamed `displayInitialContent` to `shouldDisplayInitialContent`. 
`ShowInitialContent` is an action to show the initial content. It makes sense to me

**Comment by tomivm** [comment]
> The prompt Form its blinking after change the prompt and press the NEW BOARD button. I think that is not an excepected behaviour

Yes! it's not an expected behavior. I'm preparing a PR to convert the entire prompt form to a controlled component. see: #100.

**Comment by tomivm** [comment]
Closed by error

**Comment by tomivm** [comment]
> The prompt Form its blinking after change the prompt and press the NEW BOARD button

@RodriSanchez1 this problem is solved on #101

**Pull request merged**